packages/core/types/index.js

```diff
@@ -875,6 +878,36 @@ export type TransformerResult = {|
 
 export type Async<T> = T | Promise<T>;
 
+export interface PluginLogger {
```

Comment:
Moved this type here to avoid another circular dependency. Plus, better to expose an interface to plugins rather than an implementation.